src/CustomList.php

```diff
@@ -129,6 +157,25 @@ public function add(mixed $item): void
         $this->items[] = $item;
     }
 
+    /**
+     * Remove duplicates of list
+     *
+     * @return void
+     * @throws CustomListTypeErrorException
+     */
+    public function unique(): CustomList
```

Comment:
Maybe we have some problems with this implementation because the items must be equal including all their properties, for example, if we have a custom list of objects to remove duplicates all properties must be the same, not only one of them, for example, remove duplicate items that have same value in property name.

I'm not sure if we want this behavior here. I'll test it with this example:

Example removing duplicates:

```php
    /** @test */
    public function it_should_remove_duplicate_objects()
    {
        $object1 = new stdClass();
        $object1->property = 'foo';

        $object2 = new stdClass();
        $object2->property = 'foo';

        $object3 = new stdClass();
        $object3->property = 'bar';

        $items = [$object1, $object2, $object3];

        $objectList = new class ($items) extends CustomList {
            protected function getListType(): string
            {
                return 'stdClass';
            }
        };

        $objectList = $objectList->unique();
        print_r($objectList->get());
        exit;
    }
```
OUTPUT:

```bash
Array
(
    [0] => stdClass Object
        (
            [property] => foo
        )

    [1] => stdClass Object
        (
            [property] => bar
        )

)
```

Example not removing elements with same property value:
```php
    /** @test */
    public function it_should_remove_duplicate_objects()
    {
        $object1 = new stdClass();
        $object1->property = 'foo';
        $object1->property2 = 'bar';

        $object2 = new stdClass();
        $object2->property = 'foo';

        $object3 = new stdClass();
        $object3->property = 'bar';

        $items = [$object1, $object2, $object3];

        $objectList = new class ($items) extends CustomList {
            protected function getListType(): string
            {
                return 'stdClass';
            }
        };

        $objectList = $objectList->unique();
        print_r($objectList->get());
        exit;
    }
```

Output:

```bash
Array
(
    [0] => stdClass Object
        (
            [property] => foo
            [property2] => bar
        )

    [1] => stdClass Object
        (
            [property] => foo
        )

    [2] => stdClass Object
        (
            [property] => bar
        )

)
```